cmake/modules/hunter_get_boost_libs.cmake

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2017 NeroBurner
+# All rights reserved.
+
+include(CMakeParseArguments) # cmake_parse_arguments
+
+include(hunter_test_string_not_empty)
+
+function(hunter_get_boost_libs)
+  set(function_name "hunter_get_boost_libs")
+  set(function_synopsis "${function_name}(VERSION requested_boost_version LIBS found_libs)")
+
+  # parse arguments
+  set(one_value_args VERSION LIBS)
+  cmake_parse_arguments(x "" "${one_value_args}" "" ${ARGV})
+
+  # No free arguments allowed
+  list(LENGTH x_UNPARSED_ARGUMENTS x_len)
+  if(NOT x_len EQUAL 0)
+    hunter_internal_error(
+      "'${function_name}' incorrect usage,"
+      " expected no free arguments '${x_UNPARSED_ARGUMENTS}'."
+      " Synopsis: ${function_synopsis}"
+    )
+  endif()
+
+  # check mandatory arguments
+  foreach(arg VERSION LIBS)
+    string(COMPARE EQUAL "${x_${arg}}" "" is_empty)
+    if(is_empty)
+      hunter_internal_error(
+        "'${function_name}' incorrect usage,"
+        " option '${arg}' with one argument is mandatory."
+        " Synopsis: ${function_synopsis}"
+      )
+    endif()
+
+    string(FIND "${x_${arg}}" " " arg_whitespace_position)
+    if(NOT arg_whitespace_position EQUAL -1)
+      hunter_internal_error(
+        "'${function_name}' incorrect usage,"
+        " argument '${arg}' with whitespaces not allowed."
+        " offending value '${x_${arg}}'"
+      )
+    endif()
+  endforeach()
+
+  # check boost component valid version range, add to 'boost_libs' if valid
+  # - version_since: first boost version the component is included
+  # - version_until: first boost version the component isn't included anymore
+  macro(boost_component)
```

Comment:
if creating a new module I'd like to convert the macro into a function and make it more general. I propose the following calling convention:
```
hunter_append_component(LIST boost_libs VERSION given-version COMPONENT dummy SINCE version_since UNTIL version_until)
```
`LIST` `VERSION` and `COMPONENT` are mandatory, `SINCE` and `UNTIL` are optional